test/distributed/test_c10d_nccl.py

```diff
@@ -235,45 +252,61 @@ def test_empty_tensors(self):
         pg.reduce(xs).wait()
         self.assertEqual(0, xs[0].numel())
 
-        ys = [[torch.cuda.FloatTensor([]) for _ in range(self.world_size)]]
+        ys = [[torch.FloatTensor([]).cuda(local_device_idx) for _ in range(self.world_size)]]
         pg.allgather(ys, xs).wait()
         for y in ys[0]:
             self.assertEqual(0, y.numel())
 
-        ys = [torch.cuda.FloatTensor([])]
-        xs = [[torch.cuda.FloatTensor([]) for _ in range(self.world_size)]]
+        ys = [torch.FloatTensor([]).cuda(local_device_idx)]
+        xs = [[torch.FloatTensor([]).cuda(local_device_idx) for _ in range(self.world_size)]]
         pg.reduce_scatter(ys, xs).wait()
         self.assertEqual(0, ys[0].numel())
 
     @requires_nccl()
     @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
     def test_broadcast_ops(self):
-        store = c10d.FileStore(self.file.name, self.world_size)
-        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = self._create_process_group_nccl(store, self.rank, self.world_size, self.opts())
+        rank_to_GPU = init_multigpu_helper(self.world_size, "nccl")
 
         def broadcast(xs, rootRank, rootTensor):
             opts = c10d.BroadcastOptions()
             opts.rootRank = rootRank
             opts.rootTensor = rootTensor
-            work = pg.broadcast(xs, opts)
-            work.wait()
+            fut = pg.broadcast(xs, opts).get_future()
+            fut.wait()
```

Comment:
n00b question, can you directly call .wait()?

Reply:
yeah both should work, i was trying to make it consistent with test_c10d_gloo so that we can merge it next, but it needs some additional work, I will just revert it back to directly calling wait()